src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java

```diff
@@ -397,12 +400,140 @@ public void getExtendedSearchesByMatter(String bearerToken, Integer executionId,
+        MasterIndexFieldByRecordClassSchemaResponse.MasterIndexField masterIndexField) {
+        return CreateExportBasedOnSearchResultsTableRequest.HeaderColumn.builder()
+            .masterIndexField(masterIndexField.getMasterIndexFieldId())
+            .displayName(masterIndexField.getDisplayName())
+            .propertyName(masterIndexField.getPropertyName())
+            .propertyType(masterIndexField.getPropertyType())
+            .isMasked(masterIndexField.getIsMasked())
+            .build();
+    }
+
+
     @Override
-    public ExtendedProductionsByMatterResponse getExtendedProductionsByMatter(String bearerToken, Integer executionId, UserAccountEntity userAccount) {
-        throw new NotImplementedException("Method not implemented");
+    public void getExtendedProductionsByMatter(String bearerToken, Integer executionId, UserAccountEntity userAccount) {
+        var armRpoExecutionDetailEntity = armRpoService.getArmRpoExecutionDetailEntity(executionId);
+        armRpoService.updateArmRpoStateAndStatus(armRpoExecutionDetailEntity, ArmRpoHelper.getExtendedProductionsByMatterRpoState(),
+                                                 ArmRpoHelper.inProgressRpoStatus(), userAccount);
+
+        StringBuilder errorMessage = new StringBuilder("Failure during ARM RPO Extended Productions By Matter: ");
+
+        GetExtendedProductionsByMatterRequestGenerator requestGenerator;
+        try {
+            requestGenerator = createExtendedProductionsByMatterRequest(armRpoExecutionDetailEntity.getMatterId());
+        } catch (NullPointerException e) {
+            throw handleFailureAndCreateException(errorMessage.append("Could not construct API request: ").append(e)
+                                                      .toString(), armRpoExecutionDetailEntity, userAccount);
+        }
+
+        ExtendedProductionsByMatterResponse extendedProductionsByMatterResponse;
+        try {
+            extendedProductionsByMatterResponse = armRpoClient.getExtendedProductionsByMatter(bearerToken, requestGenerator.getJsonRequest());
+        } catch (FeignException e) {
+            // this ensures the full error body containing the ARM error detail is logged rather than a truncated version
+            log.error(errorMessage.append("Unable to get ARM RPO response").toString() + " {}", e.contentUTF8());
+            throw handleFailureAndCreateException(errorMessage.toString(), armRpoExecutionDetailEntity, userAccount);
+        }
+
+        handleResponseStatus(userAccount, extendedProductionsByMatterResponse, errorMessage, armRpoExecutionDetailEntity);
+        if (isNull(extendedProductionsByMatterResponse.getProductions())
+            || CollectionUtils.isEmpty(extendedProductionsByMatterResponse.getProductions())
+            || isNull(extendedProductionsByMatterResponse.getProductions().getFirst())
+            || isNull(extendedProductionsByMatterResponse.getProductions().getFirst().getProductionId())) {
```

Comment:
As `productionId` is a String, consider using `StringUtils.isBlank()` from `org.apache.commons.lang3` rather than `isNull()`. 

`isBlank()` will catch `null`, `""` and `" "`, and I suspect any of those values would be problematic if they were to be returned in the response.

Reply:
Done.